tracer/src/Datadog.Trace/Headers/HeaderNormalizer.cs

```diff
@@ -0,0 +1,79 @@
+// <copyright file="HeaderNormalizer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Text;
+
+namespace Datadog.Trace.Headers
+{
+    internal class HeaderNormalizer : IHeaderNormalizer
```

Comment:
What was the reason for moving this to an instance instead of a static?

Reply:
To mock it in SpanContextPropagatorTests to have more unit tests there. It's nitpicking, but I assumed that for 2 additional instances it was worth it.

Reply:
Do we really need to mock it though? 🤔 We have unit tests for the normalization itself already, I'm not sure what mocking this is really achieving? I'm not fundamentally against it, this just seems like a classic "only needs to be a static function" to me.

Reply:
IMO, testing SpanContextPropagator shouldn't have to know the behavior of normalization. Here if you want to test it using a static method, you need to test both the propagator and the normalization behavior. 
Also, having the normalization in a string extension didn't really made sense to me, and it didn't really belong in SpanContextPropagator either. 

Thus it felt right to do it. But it's not a big deal either I agree and I can revert that part. We need a third party to settle the discussion :p

Reply:
I think I fall in the middle ground 😛 I agree with Andrew that we don't really need to mock it and can test the static function since the normalization strategy shouldn't change. I also agree with Pierre that this deserves to be in its own class because it was an overreach to have this logic hidden inside `StringExtensions`